### DISCUSSION TITLE
Can filter-repo keep file history for files that have been moved?

### DISCUSSION THREAD

**Discussion by RochaStratovan** [newren/git-filter-repo#362] [Q&A] [answered]
Hello,

I'm not certain if `git filter-repo` will support what I need, I know `git filter-branch` does not do it.

I've been tasked with breaking a monolithic repository into several smaller repositories. We want full commit history for the files. We don't care if the commit SHAs change. We also have many files or folders that have been moved or renamed, and we want the _**complete**_ history for a file.

Right now we can see the _complete_ (ish) history of a file using the `git log --follow` option.

Can this be done with `git filter-repo`, if so how? I'm not seeing this in the examples.

A simple setup test for this follow:
1. Create test repository
1. Add the following folders and files
   1. keep/keep_1.txt
   1. keep/keep_2.txt
   1. discard/discard_1.txt
   1. discard/discard_2.txt
1. move/rename the folder keep as keep-new

Now `git log keep-new/keep_1.txt` will have one entry, the commit for the move.

However `git log --follow keep-new/keep_1.txt` will have two entries, the move and the original add commit messages.

How would I use `git filter-repo` to remove the discard folder, maintain the keep-new folder along with the _**complete**_ history for the files in the keep-new folder?

**Comment by newren** [comment, marked as answer]
See issue #25 and issue #265.  You can achieve this, but not with some automatic command that takes care of it for you; so those other issues for pointers.
